api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -178,239 +184,973 @@ public CohortDefinition getPatientsWhoAreNotPregnantAndNotBreastfeedingD2() {
+   * <p><b>Description:</b> Number of all patients currently on ART who are included in DSD model:
+   * <b>Dispensa Semestral (DS)</b>
    *
    * @return {@link CohortDefinition}
    */
   public CohortDefinition getN3() {
     CompositionCohortDefinition cd = new CompositionCohortDefinition();
     cd.setName(
-        "N3 - Number of active patients on ART that are participating in GAAC at the end of the month prior to month of results submission deadline.");
-    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+        "N3: Number of all patients currently on ART who are included in DSD model: Dispensa Semestral (DS)");
+
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    CohortDefinition nextArtPickUpScheduledORdispensaSemestral =
+        DsdQueries
+            .getPatientsWithTypeOfDispensationOnMdcInTheMostRecentFichaClinicaOrWithPickupOnFilaBetween(
+                175, 190, Arrays.asList(hivMetadata.getSemiannualDispensation().getConceptId()));
+
+    cd.addSearch(
+        "nextArtPickUpScheduledORdispensaSemestral",
+        EptsReportUtils.map(
+            nextArtPickUpScheduledORdispensaSemestral, "endDate=${endDate},location=${location}"));
+
+    cd.addSearch(
+        "D3",
+        EptsReportUtils.map(
+            resumoMensalCohortQueries.getActivePatientsInARTByEndOfCurrentMonth(false),
+            "endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString("D3 AND nextArtPickUpScheduledORdispensaSemestral");
+
+    return cd;
+  }
+
+  /**
+   * <b>Name: N4</b>
+   *
+   * <p><b>Description:</b> N4: Number of all patients currently on ART who are included in DSD
+   * model: <b>Dispensa Anual (DA)</b>
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition getN4() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.setName(
+        "N4: Number of all patients currently on ART who are included in DSD model: Dispensa Anual (DA)");
+
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    CohortDefinition nextArtPickUpScheduledORdispensaAnual =
+        DsdQueries
+            .getPatientsWithTypeOfDispensationOnMdcInTheMostRecentFichaClinicaOrWithPickupOnFilaBetween(
+                335,
+                395,
+                Arrays.asList(hivMetadata.getAnnualArvDispensationConcept().getConceptId()));
+
+    cd.addSearch(
+        "nextArtPickUpScheduledORdispensaAnual",
+        EptsReportUtils.map(
+            nextArtPickUpScheduledORdispensaAnual, "endDate=${endDate},location=${location}"));
+
+    cd.addSearch(
+        "D3",
+        EptsReportUtils.map(
+            resumoMensalCohortQueries.getActivePatientsInARTByEndOfCurrentMonth(false),
+            "startDate=${endDate},endDate=${endDate},location=${location}"));
```

Comment:
Yes it is ok.
That's because the method getActivePatientsInARTByEndOfCurrentMonth from resumoMensalCohortQueries is mapped to receive startDate and endDate.
Yet because of the (false) parameter, startDate won't be used, it's there just to give something to that method as it waits for both startDate and endDate.